spec/hardware/disk.fmf

```diff
@@ -9,20 +9,32 @@ description: |
            # Number or string, the amount of storage requested.
            # Bytes are assumed when no unit is specified.
          - size: 1234|">= 512 GiB"
+           # String, disk model requested.
+         - model-name: "well-known disk model"
+
+    .. versionchanged:: 1.32
+       Added `model-name` into specification.
 
 example:
   - |
     # Require a disk big enough for testing
     disk:
       - size: 500 GB
 
+  - |
+    # Require a disk with specified model
+    disk:
+      - model-name: 'PERC H310'
+
   - |
     # Multiple disks can be requested as well
     disk:
       - size: '>= 2 GB'
-      - size: '>= 20 GB'
+      - model-name: 'PERC H310'
 
 link:
+  - https://github.com/teemtee/tmt/pull/2705
+  - https://gitlab.com/testing-farm/artemis/-/merge_requests/1195
```

Comment:
Do we want to include these links in the [online documentation](https://tmt--2708.org.readthedocs.build/en/2708/spec/hardware.html#disk)? Are they interesting enough for the user?

Reply:
> Thanks for preparing this.

Thanks for your review.

> Do we want to include these links in the [online documentation](https://tmt->-2708.org.readthedocs.build/en/2708/spec/hardware.html#disk)? Are they interesting enough for the user?

I guess the answer is NO, so that two merge requests should be merged first, then I'm gonna to update this merge request with the two links removed.So,would you please spare a little time to review https://github.com/teemtee/tmt/pull/2705 (which is clean and tiny)? Thanks^^

Reply:
Chicken and egg, I'm afraid. The tmt PR looks small, true, but there's no way to get the implementation into Artemis quickly enough to allow this solution.

Also, note the existing links:

```yaml
link:
  - implemented-by: /tmt/steps/provision/artemis.py
  - implemented-by: /tmt/steps/provision/mrack.py
  - implemented-by: /tmt/steps/provision/testcloud.py
```

By addition of `model-name`, all of them need modification as well, to state that only `size` is supported.

All in all, I would propose this to be the state of `link`:

```yaml
link:
  - implemented-by: /tmt/steps/provision/artemis.py
    note: "``size`` only"
  - implemented-by: /tmt/steps/provision/mrack.py
    note: "``size`` only"
  - implemented-by: /tmt/steps/provision/testcloud.py
    note: "``size`` only with ``=``, ``>=`` and ``<=`` operators only"
```

And we would undo the `note` in `artemis` and `mrack` as we add the implementations. No chance for confusion if we get delayed at any point.

Reply:
Ack, sounds good.